File: examples/publisher/ExamplePublisher.java
package examples.publisher;

import com.roslib.ros.NodeHandle;
import com.roslib.ros.Publisher;
import com.roslib.tinyros_hello.TinyrosHello;

public class ExamplePublisher {

    public static void main(String[] args) throws InterruptedException {
        NodeHandle nh = new NodeHandle();

        while (!nh.initNode("127.0.0.1")) {
            System.out.println("Java: initNode failed.");
            Thread.sleep(500);
        }

        TinyrosHello msg = new TinyrosHello();
        Publisher<TinyrosHello> pub = new Publisher<>("tinyros_hello", msg);

        nh.advertise(pub);

        while(nh.ok()) {
            if (pub.negotiated()) {
                msg.hello = "Hello, tiny-ros ^_^";
                pub.publish(msg);
            }
            
            nh.spinOnce();
            Thread.sleep(500);
        }
    }
}

File: examples/publisher/ExamplePublisher.cpp
/*
 * File      : ExamplePublisher.cpp
 * This file is part of tiny_ros
 *
 * Change Logs:
 * Date           Author       Notes
 * 2018-04-24     Pinkie.Fu    initial version
 */
#include "tiny_ros/ros.h"
#include "tiny_ros/tinyros_hello/TinyrosHello.h"

int main (int argc, char *argv[]) {
  tinyros::init("127.0.0.1");
  tinyros::Publisher hello_pub ("tinyros_hello", new tinyros_hello::TinyrosHello());
#if 1
  tinyros::nh()->advertise(hello_pub);
#else
  tinyros::udp()->advertise(hello_pub);
#endif
  while (true) {
    tinyros_hello::TinyrosHello msg;
    msg.hello = "Hello, tiny-ros ^_^ ";
    hello_pub.publish (&msg);
#ifdef WIN32
    Sleep(1000);
#else
    sleep(1);
#endif
  }
  return 0;
}

